internal/dockercompose/event.go

```diff
@@ -30,79 +30,122 @@ func EventFromJsonStr(j string) (Event, error) {
 }
 
 // https://docs.docker.com/engine/reference/commandline/events/
-type Type string
+type Type int
 
 const (
-	// Add 'types' here (and to `UnmarshalJSON` below) as we support them
-	TypeContainer Type = "container"
+	// Add 'types' here (and to `stringToType` below) as we support them
+	TypeUnknown Type = iota
+	TypeContainer
 )
 
+var stringToType = map[string]Type{
+	"container": TypeContainer,
+}
+
+func (t Type) String() string {
+	for str, typ := range stringToType {
```

Comment:
Why not make it a type to string map, and then do a lookup in to it rather than iterating through the entire map?